### DISCUSSION TITLE
Understanding and resolving Let's Encrypt failure

### DISCUSSION THREAD

**Discussion by alimanfoo** [nebari-dev/nebari#710] [Q&A]
I have a qhub deployment where I have destroyed and then redeployed. It's configured to use Let's Encrypt, and on the first deployment everything worked fine, but now the second (re)deployment I'm getting NET::ERR_CERT_AUTHORITY_INVALID when I try to access in my browser. I'm guessing that for some reason during the second deployment the Let's Encrypt process to obtain a certificate failed, although I don't really have a good understanding of how Let's Encrypt works. I'm also guessing that perhaps this might be because the update to the DNS record hadn't propagated at the time that the certificate was being obtained, although that is a wild guess.

I'm just wondering if anyone could explain a little more about how obtaining a certificate via Let's Encrypt works, in particular at what point during the deployment process is the certificate obtained, and is there any way to somehow rerun or retry the certificate generation after deployment in case the certificate process failed?

Sorry if the questions doesn't make any sense, way out of my depth here!

**Comment by alimanfoo** [comment]
Cross-referencing this issue which looks relevant: https://github.com/Quansight/qhub/issues/683

**Comment by viniciusdc** [comment]
Hi @alimanfoo sorry for the long delay with a proper answer, have you encountered that problem again, after [`24adfce`](https://github.com/Quansight/qhub/commit/24adfce090f3f4a46863bb4e6151de2e608c99ea) ?

**Comment by Adam-D-Lewis** [comment]
I'm closing old discussions in a repo cleanup effort.  Feel free to re-open if needed.
